Source/DafnyCore/Compilers/SinglePassCompiler.cs

```diff
@@ -64,6 +65,23 @@ public abstract class SinglePassCompiler : Plugins.Compiler {
 
     public CoverageInstrumenter Coverage;
 
+    public virtual int RunProcess(ProcessStartInfo psi, Process process, String platform, TextWriter outputWriter) {
```

Comment:
I think this API could be a bit cleaner: how about always passing in a process `p` and using `p.Start()` if it's not started yet, instead of passing both `psi` and `process`?

Reply:
I was suggesting to pass just `process` and omit psi; wouldn't that work?  You can access the startinfo from the `process.StartInfo` property: https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.process.startinfo?view=net-6.0

Reply:
API refactored and simplified.